src/main/java/de/zalando/aruha/nakadi/partitioning/OrderingKeyFieldsPartitioningStrategy.java

```diff
@@ -0,0 +1,71 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONObject;
+
+import java.util.List;
+
+import static java.lang.Math.abs;
+
+public class OrderingKeyFieldsPartitioningStrategy implements PartitioningStrategy {
+
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final int numberOfPartitions) throws InvalidOrderingKeyFieldsException {
```

Comment:
As for me - the implementation is too kafka-specific (assuming that partition is a number from 0 to partitionsNum - 1).
I think we should abstract from that with a help of some new method of TopicRepository like getPartitionByIndex.